enum.c

```diff
@@ -2742,20 +2742,28 @@ each_with_index_i(RB_BLOCK_CALL_FUNC_ARGLIST(i, memo))
 
 /*
  *  call-seq:
- *     enum.each_with_index(*args) { |obj, i| block } ->  enum
- *     enum.each_with_index(*args)                    ->  an_enumerator
+ *    each_with_index {|element, i| ..... } -> self
+ *    each_with_index                       -> enumerator
  *
- *  Calls <em>block</em> with two arguments, the item and its index,
- *  for each item in <i>enum</i>.  Given arguments are passed through
- *  to #each().
+ *  With a block given, calls the block with each element and its index;
+ *  returns +self+:
  *
- *  If no block is given, an enumerator is returned instead.
+ *    h = {}
+ *    (1..4).each_with_index {|element, i| h[element] = i } # => 1..4
+ *    h # => {1=>0, 2=>1, 3=>2, 4=>3}
  *
- *     hash = Hash.new
- *     %w(cat dog wombat).each_with_index { |item, index|
- *       hash[item] = index
- *     }
- *     hash   #=> {"cat"=>0, "dog"=>1, "wombat"=>2}
+ *    h = {}
+ *    %w[a b c d].each_with_index {|element, i| h[element] = i }
+ *    # => ["a", "b", "c", "d"]
+ *    h # => {"a"=>0, "b"=>1, "c"=>2, "d"=>3}
+ *
+ *    a = ()
```

Comment:
This example section looks wrong.  I think `a = ()` should be `h1 ={}` and `h` should be renamed to `h0`.  However, it would be best to check that it works by running it.

Reply:
Just needed <tt>a = ()</tt> removed, I think.

Reply:
I don't think so.  Where do `h0` and `h1` come from in the example following this code?

Reply:
Ook.  Ran in irb with them previously defined.  Will review.

Reply:
Changed collector to a = [], to avoid confusion among hashes.

Reply:
I think fixed.